content/developer/reference/frontend/unit_testing/mock_server.rst

```diff
@@ -0,0 +1,386 @@
+                    <field name="display_name" />
+                </list>
+            `,
+        };
+    }
+
+    defineModels({ ResPartner });
+
+This code will make these data available for *all* tests in the current test file.
+Of course, defining a class and calling `defineModels` can also be done from *within*
+a given test to limit the scope of that model to the current test.
+
+Other methods such as `defineMenus`, `defineActions` or `defineParams` can also
+be used to configure the current mock server. Most of their API is quite straightforward
+(i.e. they receive JSON-like descriptions of menus, actions, etc.).
+
+Mock models: requests
+---------------------
+
+Many test cases only require one or a few mock models to work. But sometimes,
+it is either too bothersome to implement the mocking logic within a model, or a
+*route* (i.e. server request URL) is simply not associated to a Python model at all.
+
+In such cases, the `onRpc` method is to be called, to associate a route or an ORM
+method to a callback.
+
+.. note::
+    Multiple `onRpc` calls can be associated to the same route / ORM method;
+    in which case they will be called sequentially from last to first defined.
+    Returning a *non-null-or-undefined* value will interrupt the current chain,
+    and return that value as final result of the server request.
+
+It can be used in 4 different ways:
+
+`onRpc`: with a route (`"/"`)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When the first argument is a `string` starting with a `"/"`, the callback
+is expected to be a *route* callback, receiving a `Request_`
+object:
+
+.. code-block:: javascript
+
+    onRpc("/route/to/test", async (request) => {
+        const { ids }  = await request.json();
+        expect.step(ids);
+        return {};
+    });
+
+By default, the return value of these callbacks are wrapped within the `body`
+of a mock `Response_` object.
+
+This is fine for most use-cases, but sometimes the callback needs to respond with
+a `Response_` object with custom `status` or `headers`.
+
+In such cases, an *optional* dictionary can be passed as a 3rd argument to specify
+whether the callback is to be considered *"pure"*, meaning that its return value
+should be returned as-is to the server caller:
+
+.. code-block:: javascript
+
+    onRpc(
+        "/not/found",
+        () => new Response("{}", { status: 404 }),
+        { pure: true }
+    );
+
+.. note::
+    Using *"pure"* request callbacks can also be used to return anything else than
+    a `Response_` object, in which case the returned value will still be wrapped
+    in the body of a mock `Response_` to comply with the `fetch_` / `XMLHttpRequest_`
+    APIs.
+
+`onRpc`: with method name(s)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When:
+
+- the first argument is a `string` *NOT* starting with a `"/"` or a list of `strings`;
+
+=> the callback is expected to be an ORM callback, only called when the request's
+   `method` matches the one given as argument.
```

Comment:
<img width="873" height="714" alt="Image" src="https://github.com/user-attachments/assets/483e1136-ce27-4405-9fed-99ce0ab74fac" />

Sphinx doesn't seem to like the `=>` since it doesn't highlight the next line content the same way.
Not sure it can be fixed while keeping the `=>` though. Maybe you could change it to some kind of boxed display, like a `tip` or `warning` ?